### ISSUE TITLE
Traversable layer of abstraction over FS APIs from Sys module

### ISSUE THREAD

**Issue by rexim** [rexim/todo#8]
- Split from #5 

---

Right now, any possible implementations of `file_stream_of_dir_tree` and `root_of_git_repo` will be coupled with FS API from `Sys` module, which makes them really difficult to unit test.

We propose to implement a mockable API over `Sys`, which consist of the following core type:

```ocaml
type node =
  | DirNode of (string * node Stream.t)
  | FileNode of string
```

Such type is really simple to traverse and mock.

Once the API is done, we need to implement `file_stream_of_dir_tree` and `root_of_git_repo` functions using it.

**Comment by rexim** [comment]
Superseded by #11 because of https://github.com/rexim/todo/pull/10#issuecomment-314656163
